moon/controller_round3.py

```diff
@@ -544,16 +562,17 @@ def pol2cart(rho, phi):
                     self.centering = False
 
                 # if seeing basemarker and homebase center is not straight ahead OR if looking past homebase in one of the directions, turn in place to adjust
+                circle_direction = 1 if self.going_around_count % 2 == 0 else -1
                 if (self.currently_following_object['object_type'] == 'basemarker' and homebase_cy < -0.2) or left_dist > 10:
                     self.centering = True
                     self.publish("desired_movement", [0, -9000, -SPEED_ON])
                 elif (self.currently_following_object['object_type'] == 'basemarker' and homebase_cy > 0.2) or right_dist > 10:
                     self.centering = True
                     self.publish("desired_movement", [0, -9000, SPEED_ON])
                 elif left_dist < 1.5 or right_dist < 1.5:
-                    self.publish("desired_movement", [float("inf"), -9000, -SPEED_ON])
+                    self.publish("desired_movement", [GO_STRAIGHT, -9000, -SPEED_ON])
                 elif left_dist > HOMEBASE_KEEP_DISTANCE + 1 or right_dist > HOMEBASE_KEEP_DISTANCE + 1:
-                    self.publish("desired_movement", [float("inf"), -9000, SPEED_ON])
+                    self.publish("desired_movement", [GO_STRAIGHT, -9000, SPEED_ON])
```

Comment:
:) good, I am glad for this `GO_STRAIGHT` constant, thanks (BTW what -9000 then means for going straight?)

Reply:
This is view angle in degrees*100 relative to movement direction of the robot. If it's 0, the camera looks forward. If it's 9000 or -9000, the camera looks to the side. It should support any angle really (e.g., look 45 degrees away from the direction of drive) but I was unable to come up with the math formulas that would turn all the 4 wheels correctly given an arbitrary angle. Maybe someone else can :) So only -9000,0,9000 have meaning for now.